src/vonage/number_insight_v2.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING, Union
+
+from .errors import NumberInsightV2Error
+from ._internal import validate_phone_number
+
+if TYPE_CHECKING:
+    from vonage import Client
+
+
+class NumberInsightV2:
+    def __init__(self, client: Client):
+        self._client = client
+        self._auth_type = 'header'
+
+    def fraud_check(self, number: str, insights: Union[str, list]):
+        validate_phone_number(number)
+        params = {'type': 'phone', 'phone': number}
+
+        if type(insights) == str:
+            self._validate_insight(insights)
+            params['insights'] = [insights]
+        elif type(insights) == list:
+            for insight in insights:
+                self._validate_insight(insight)
+            params['insights'] = insights
```

Comment:
The type isn't forced to be a string or list, that's just a type hint, but I've taken on board what you've suggested and rewritten the method, plus added a check in case a str or list isn't what's passed in.